Neo4j.Driver/Neo4j.Driver/Internal/Routing/ClusterConnection.cs

```diff
@@ -47,7 +48,35 @@ public override void OnError(Exception error)
             {
                 _errorHandler.OnConnectionError(_uri, error);
             }
-            else if (error.IsClusterError())
+            else
+            {
+                HandleClusterError(error);
+            }
+            throw error;
+        }
+
+        public async Task OnErrorAsync(Exception error)
```

Comment:
@ali-ince This is the method that need to be plugged in `DelegatedSocketConnection`

Reply:
I think adding `DelegatedConnection.OnErrorAsync` makes sense. The only actual place where async job is done seems to be `ClusterConnection` though. This also brings another option of introducing `IAsyncErrorHandler` interface with `OnErrorAsync`  and selectively calling it from within `DelegatedConnection.TaskWithErrorHandling`?